server/s3_object_action_base.cc

```diff
@@ -106,33 +106,42 @@ void S3ObjectAction::fetch_object_info() {
 }
 
 void S3ObjectAction::fetch_object_info_success() {
+  s3_log(S3_LOG_DEBUG, request_id, "%s Entry\n", __func__);
   request->set_object_size(object_metadata->get_content_length());
   // TODO: Read extended object's parts/fragments, depending on the type of
   // primary object.
   // If object is extended, create S3ObjectExtendedMetadata and load extended
   // entries.
   if (object_metadata->is_object_extended()) {
-    // Read the extended parts of the object
-    extended_obj_metadata =
+    // Read the extended parts of the object from extended index table
+    std::shared_ptr<S3ObjectExtendedMetadata> extended_obj_metadata =
         object_metadata_factory->create_object_ext_metadata_obj(
             request, request->get_bucket_name(), request->get_object_name(),
             object_metadata->get_obj_version_key(),
             object_metadata->get_number_of_parts(),
-            object_metadata->get_number_of_fragments(), object_list_oid);
-
+            object_metadata->get_number_of_fragments(),
+            bucket_metadata->get_extended_metadata_index_oid());
+    object_metadata->set_extended_object_metadata(extended_obj_metadata);
     extended_obj_metadata->load(
         std::bind(&S3ObjectAction::fetch_ext_object_info_success, this),
         std::bind(&S3ObjectAction::fetch_ext_object_info_failed, this));
   } else {
     next();
   }
+  s3_log(S3_LOG_DEBUG, request_id, "%s Exit\n", __func__);
 }
 
-void S3ObjectAction::fetch_ext_object_info_success() { next(); }
+void S3ObjectAction::fetch_ext_object_info_success() {
+  s3_log(S3_LOG_DEBUG, request_id, "%s Entry\n", __func__);
+  next();
+  s3_log(S3_LOG_DEBUG, request_id, "%s Exit\n", __func__);
+}
 
 void S3ObjectAction::fetch_ext_object_info_failed() {
   // TBD: Add code to handle failure in loading extended entries.
+  s3_log(S3_LOG_DEBUG, request_id, "%s Entry\n", __func__);
```

Comment:
Failure scenario will be handled separately in which ticket ?

Reply:
need to implement in derived class...will do this soon.

Reply:
done. Pushed changes